src/main/java/attendance/domain/AttendanceHistory.java

```diff
@@ -1,49 +1,82 @@
-    private final AttendanceType attendanceType;
+    private final Set<Attendance> attendances = new HashSet<>();
+
+    public Attendance addAttendance(final Attendance attendance) {
+        validateDuplicatedAttendance(attendance.getDate());
+        attendances.add(attendance);
+        return attendance;
+    }
 
-    public AttendanceHistory(LocalDateTime attendanceDateTime, AttendanceType attendanceType) {
-        this.attendanceDateTime = attendanceDateTime;
-        this.attendanceType = attendanceType;
+    public void validateDuplicatedAttendance(final LocalDate attendanceDate) {
+        findAttendance(attendanceDate).ifPresent(attendance -> {
+            throw new IllegalArgumentException(DUPLICATED_ATTENDANCE.getMessage());
+        });
     }
 
-    public LocalDateTime getAttendanceDateTime() {
-        return attendanceDateTime;
+    public Optional<Attendance> findAttendance(final LocalDate date) {
+        return attendances.stream()
+                .filter(attendance -> attendance.isDateEquals(date))
+                .findAny();
     }
 
-    /**
-     * TODO
-     * 내부에서 boolean을 반환할 수 있겠다
-     */
-    public AttendanceType getAttendanceType() {
-        return attendanceType;
+    public Attendance modifyAttendance(final Attendance modifiedAttendance) {
+        LocalDate attendanceDateToModify = modifiedAttendance.getDate();
+        findAttendance(attendanceDateToModify).ifPresentOrElse(attendances::remove, () -> {
+            throw new IllegalArgumentException(NO_ATTENDANCE_TO_MODIFY.getMessage());
+        });
+        return addAttendance(modifiedAttendance);
     }
 
-    public boolean isAttendanceDateEquals(LocalDate date) {
-        LocalDate attendanceDate = attendanceDateTime.toLocalDate();
-        return attendanceDate.equals(date);
+    public List<Attendance> getMonthlyAttendances(final LocalDate today) {
+        return attendances.stream()
+                .filter(attendance -> attendance.isYearMonthEquals(today))
+                .toList();
     }
 
-    @Override
-    public boolean equals(Object o) {
-        if (this == o) {
-            return true;
+    public AttendanceStatistics getAttendanceStatistics(final LocalDate today) {
+        Map<AttendanceStatus, Integer> statistics = new HashMap<>();
+        initializeStatistics(statistics);
+        for (int i = today.getDayOfMonth() - 1; i > 0; i--) {
+            LocalDate date = today.minusDays(i);
+            addAttendanceStatusCount(date, statistics);
         }
-        if (o == null || getClass() != o.getClass()) {
-            return false;
+        return new AttendanceStatistics(statistics.get(ATTENDANCE), statistics.get(LATE), statistics.get(ABSENCE));
+    }
+
+    private void addAttendanceStatusCount(final LocalDate date, final Map<AttendanceStatus, Integer> statistics) {
+        boolean isOperationDate = CampusManager.isOperationDate(date);
+        if (!isOperationDate) {
+            return;
+        }
```

Comment:
```suggestion
        if (!CampusManager.isOperationDate(date)) {
            return;
        }
```

요게 가독에 더 좋겠네요. 메서드내 변수들도 메서드내에서 따로 쓰이지않는다면 굳이 추출할 필요는 없는거같아요